javascript/src/main/components/RoomSlots/RoomSlotForm.js

```diff
@@ -0,0 +1,185 @@
+import React, { useState, useRef } from "react";
+import { Form, Button, Row, Col } from "react-bootstrap";
+import {checkTime, checkFilled, checkCourseQuarter} from "main/utils/FormHelpers";
+import { formatTimeToLocalTime } from "main/utils/RoomSlotFormHelpers";
+
+const RoomSlotForm = ({ createRoomSlot/*, updateRoomSlot, existingRoomSlot*/}) =>{
+    const locationRef = useRef(null);
+    const startTimeRef = useRef(null);
+    const endTimeRef = useRef(null);
+    const quarterRef = useRef(null);
+    
+    const emptyRoomSlot = {
+        location: "",
+        quarter: "",
+        dayOfWeek: "Monday",
+        startTime:"",
+        endTime:""
+    }
+
+    const [roomSlot, setRoomSlot] = useState(emptyRoomSlot);
+
+    const handleOnSubmit = (e) => {
+        e.preventDefault();
+        var isValid = checkInputs();
+
+        if(isValid){
+            let quarterNum = "1";
+            if(roomSlot.quarter.substr(0, 1)==="S"){
+                quarterNum =  "2";
+            }
+            else if(roomSlot.quarter.substr(0, 1)==="M"){
+                quarterNum =  "3";
+            }
+            else if(roomSlot.quarter.substr(0, 1)==="F"){
+                quarterNum =  "4";
+            }
+            let formattedQuarter = "20" + roomSlot.quarter.substr(1, 2) + quarterNum;
```

Comment:
This logic can probably be refactored a bit and moved with the rest of the quarter format utilities.